lib/rdoc/ri/driver.rb

```diff
@@ -1152,13 +1152,36 @@ def interactive
   # Is +file+ in ENV['PATH']?
 
   def in_path? file
-    return true if file =~ %r%\A/% and File.exist? file
+    extensions = [''].concat((windows? ? ENV['PATHEXT']&.split(File::PATH_SEPARATOR) : nil).to_a)
```

Comment:
`IO.popen` supports only executable by itself (".exe", ".com") and by `cmd.exe` (".cmd" and ".bat").
Others in `PATHEXT` by default, ".js", ".vbs", and so on, which need extra registry handling are not supported.

Reply:
I'm realizing how difficult it is to take all cases into account...
I took a journey to win32.c and dln_find.c. On windows, IO.popen("commandline", "w") not always uses shell　I understood, is that correct?
It seems to be reasonable to use RbConfig::CONFIG["EXECUTABLE_EXTS"] instead of ENV['PATHEXT'], It's been prepared that way. I try it.